### PR TITLE
Support for deploying

### LINKED ISSUE #32: Deploy script to build the frontend
Deploy script to build the frontend.

Idea: let flask be the server for the compiled static page (in addition to serving the dynamic api endpoints). It would be nice if the flask dockerfile can be reused for development and deployed serving.

### PR DESCRIPTION
**Pull request by pontus**
Changes to build production images while supporting developer workflow.

Fixes #32.

### CODE REVIEW

File: system/encam.standalone.nginx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        listen 8443 ssl http2;
+        listen 8080;
+
+        ssl_certificate /code/cert.pem;
+        ssl_certificate_key /code/key.pem;
+        ssl_dhparam /code/dhparam.pem;
+        root /app/dist/browser/;
+
+        location / {
+            try_files $uri $uri/ @wsgi =415;
```
Comment (danr) [inline system/encam.standalone.nginx]: What's 415?

File: api/Dockerfile [hunk 2]
```diff
@@ -1,30 +1,10 @@
...
 
 RUN mkdir /code
 
-COPY ./api/requirements.txt /code/
-COPY ./api/SIM.csv .
-
-RUN pip3 install -r /code/requirements.txt
-
+WORKDIR /api_source
+RUN ls
```
Comment (danr) [inline api/Dockerfile]: Why do we have `RUN ls` here?

File: docker-compose.yml [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
-    nginx:
-        build: ./nginx
-        container_name: nginx
-        restart: always
-        ports:
-            - 80:80
-    frontend:
+            - type: bind
+              source: ./api
+              target: /api_source
```
Comment (danr) [inline docker-compose.yml]: Slightly confusing that it is not the **source** but the **target** that is api_**source**. What about `target: /api_src`?

File: system/nginx.conf [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+               application/x-javascript application/xml
+               application/atom+xml text/javascript;
+
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header X-Forwarded-Proto $scheme;
+    proxy_set_header Host $http_host;   
+
+    include encam;
+
+}
```
Comment (danr) [inline system/nginx.conf:37]: Now where is both `system/nginx.conf` and `nginx/nginx.conf`. The `nginx` directory can now be removed? It also includes a `nginx/Dockerfile` which seems unused.

File: system/supervisord.conf [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+serverurl=unix:///tmp/supervisor.sock         ; use a unix:// URL  for a unix socket
+
+[program:nginx]
+command=/usr/sbin/nginx
+stderr_logfile = /dev/stdout
+stdout_logfile = /dev/stderr
+stdout_logfile_maxbytes = 0
+stderr_logfile_maxbytes = 0
+
+[program:uwsgi1]
```
Comment (danr) [inline system/supervisord.conf]: Any particular reason this is `uwsgi1` instead of `uwsgi`?

File: system/nginx.conf (hunk 4 above)
Comment (pontus) [reply system/nginx.conf:37]: Removed.

File: docker-compose.yml (hunk 3 above)
Comment (pontus) [reply docker-compose.yml]: Sure, fixed.

File: api/Dockerfile (hunk 2 above)
Comment (pontus) [reply api/Dockerfile]: Fixed.

File: system/encam.standalone.nginx (hunk 1 above)
Comment (pontus) [reply system/encam.standalone.nginx]: Unsupported media type, but mostly this is my eyes being not noticing when checking through before push.

File: system/supervisord.conf (hunk 5 above)
Comment (pontus) [reply system/supervisord.conf]: I had several at first but removed the others. Fixed.

**Comment by danr** [comment]
Something interestesting happened with the history when the base branch was changed so it's now non-trivial to rebase this on `dev`. One solution could be to squash all the commits here. Anyone against?

**Comment by danr** [comment]
Seems to be easier to make a merge commit merging dev into this branch. The history gets a bit wonky but I don't think it'll matter.

**Comment by pontus** [comment]
Hrrm, not sure I understand the issue with merging to dev, but I don't think there is any loss in losing these commits, so squashing would work? (UI is not intuitive to me, but the arrow besides `Merge pull request` allows selection of other options.

**Pull request merged**